### PR TITLE
feat: Update to Gatsby Starter

### PR DESCRIPTION
**Pull request by kasperkristensen**

### CODE REVIEW

**Review by olivermrbl — CHANGES_REQUESTED** [review]
Two bigger things, that we could start out with:
- Let's rename all files to a `kebab-cased` version
- Let's try to include our `medusa-js` client instead of the `services/api.js`. You can install the latest version by doing `yarn add @medusajs/medusa-js@canary`.

The client should be initialised by creating a util function like so: 
```
import Medusa from "@medusajs/medusa-js"

const BACKEND_URL = process.env.GATSBY_STORE_URL || "http://localhost:4000"

export const createClient = () => new Medusa({ baseUrl: BACKEND_URL })
```

But feel free, to do it your preferred way :)

**Comment by kasperkristensen** [comment]
> Two bigger things, that we could start out with:
> 
> * Let's rename all files to a `kebab-cased` version
> * Let's try to include our `medusa-js` client instead of the `services/api.js`. You can install the latest version by doing `yarn add @medusajs/medusa-js@canary`.
> 
> The client should be initialised by creating a util function like so:
> 
> ```
> import Medusa from "@medusajs/medusa-js"
> 
> const BACKEND_URL = process.env.GATSBY_STORE_URL || "http://localhost:4000"
> 
> export const createClient = () => new Medusa({ baseUrl: BACKEND_URL })
> ```
> 
> But feel free, to do it your preferred way :)

Made the requested changes, have a look :)

File: src/components/cartView/cart-view.jsx [hunk 1]
```diff
@@ -0,0 +1,127 @@
+import React, { useContext } from "react";
+import DisplayContext from "../../context/display-context";
+import StoreContext from "../../context/store-context";
+import { Link, navigate } from "gatsby";
+import * as styles from "../../styles/CartView.module.css";
+import { quantity, sum, formatPrice } from "../../utils/helperFunctions";
```
Comment (olivermrbl) [inline src/components/cartView/cart-view.jsx]: Should be `helper-functions`

File: src/components/cartView/cart-view.jsx [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+      </div>
+      <div className={styles.bottom}>
+        <button
+          className={styles.checkoutbtn}
+          onClick={() => {
+            updateCheckoutStep(1);
+            updateCartViewDisplay();
+            navigate("/checkout");
+          }}
+          disabled={cart.items.length < 1 ? true : ""}
```
Comment (olivermrbl) [inline src/components/cartView/cart-view.jsx]: Why not `false`?

File: src/components/checkout/checkout-summery.jsx [hunk 3]
```diff
@@ -0,0 +1,100 @@
+import React, { useContext } from "react";
+import { PuffLoader } from "react-spinners";
+import * as styles from "../../styles/CheckoutSummery.module.css";
+import * as itemStyles from "../../styles/CartView.module.css";
+import { Link } from "gatsby";
+import { formatPrice } from "../../utils/helperFunctions";
+import { sum, quantity } from "../../utils/helperFunctions";
+import DisplayContext from "../../context/display-context";
+
+const CheckoutSummery = ({ cart }) => {
```
Comment (olivermrbl) [inline src/components/checkout/checkout-summery.jsx]: Rename to `CheckoutSummary`

File: src/components/checkout/information-step.jsx [hunk 4]
```diff
@@ -0,0 +1,143 @@
...
+    <div style={{ flexGrow: "1" }}>
+      <h2>Address</h2>
+      <Formik
+        initialValues={{
+          first_name: savedValues.first_name || "",
+          last_name: savedValues.last_name || "",
+          email: savedValues.email || "",
+          address_1: savedValues.address_1 || "",
+          address_2: savedValues.address_2 || "",
+          country_code: "Denmark",
```
Comment (olivermrbl) [inline src/components/checkout/information-step.jsx]: Should not be fixed. Please use country from region.

File: src/components/checkout/injectable-card.jsx [hunk 5]
```diff
@@ -0,0 +1,78 @@
+import React, { useContext, useState } from "react";
+import { CardElement, useStripe, useElements } from "@stripe/react-stripe-js";
+import { navigate } from "gatsby";
+import StoreContext from "../../context/store-context";
+import DisplayContext from "../../context/display-context";
+import * as styles from "../../styles/InjectableCardForm.module.css";
+import { BiLeftArrowAlt } from "react-icons/bi";
+
+const InjectableCard = () => {
```
Comment (olivermrbl) [inline src/components/checkout/injectable-card.jsx]: Rename to `InjectablePaymentCard` 👍

File: src/components/checkout/payment-step.jsx [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+import StoreContext from "../../context/store-context";
+import InjectableCard from "./injectable-card";
+import getStripe from "../../utils/stripe";
+
+const PaymentStep = () => {
+  const { createPaymentSession } = useContext(StoreContext);
+
+  useEffect(() => {
+    createPaymentSession();
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```
Comment (olivermrbl) [inline src/components/checkout/payment-step.jsx]: Would it have any negative consequences to simply include it as dependency?

File: src/context/display-context.js [hunk 7]
```diff
@@ -0,0 +1,57 @@
+import React, { useReducer } from "react";
+
```
Comment (olivermrbl) [inline src/context/display-context.js:2]: General comment: Rename everything from summery -> summary 👍

File: src/pages/product/[id].js [hunk 8]
```diff
@@ -0,0 +1,137 @@
...
+  }, [product]);
+
+  const handleQtyChange = (action) => {
+    if (action === "inc") {
+      if (options.quantity < 10)
+        setOptions({
+          variantId: options.variantId,
+          quantity: options.quantity + 1,
+          size: options.size,
+        });
```
Comment (olivermrbl) [inline src/pages/product/[id].js]: Not sure, that I understand this condition. Please elaborate :)

**Review by olivermrbl — CHANGES_REQUESTED** [review]
Some additional minor changes 😄 

And let's also rename all styling files to `kebab-case`

File: src/components/cartView/cart-view.jsx (hunk 2 above)
Comment (kasperkristensen) [reply src/components/cartView/cart-view.jsx]: Old habbit, but changed now :)

File: src/components/checkout/payment-step.jsx (hunk 6 above)
Comment (kasperkristensen) [reply src/components/checkout/payment-step.jsx]: It creates a loop

File: src/pages/product/[id].js (hunk 8 above)
Comment (kasperkristensen) [reply src/pages/product/[id].js]: Changed it to allow it as long as the quantity is less than inventory :)

File: src/pages/index.js [hunk 9]
```diff
@@ -1,240 +1,137 @@
...
+                viewBox="0 0 24 24"
+                height="1em"
+                width="1em"
+                xmlns="http://www.w3.org/2000/svg"
+              >
+                <path d="M12 4l-1.41 1.41L16.17 11H4v2h12.17l-5.58 5.59L12 20l8-8z"></path>
+              </svg>
+            </a>
+            <a
+              href="https://github.com/medusajs/nextjs-starter-medusa"
```
Comment (olivermrbl) [inline src/pages/index.js:93]: Change to: `https://github.com/medusajs/gatsby-starter-medusa`

**Review by olivermrbl — CHANGES_REQUESTED** [review]
A final, final comment in regards to the styling 😄  
Let's also lowercase all the styling files, that was not `kebab-cased` in previous commit.

image

**Review by olivermrbl — APPROVED** [review]
Great job!! Let's ship it 🚀

**Pull request merged**